data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/PostProposerSlashing.java

```diff
@@ -29,23 +30,38 @@
 import io.javalin.plugin.openapi.annotations.OpenApiResponse;
 import tech.pegasys.teku.api.DataProvider;
 import tech.pegasys.teku.api.NodeDataProvider;
-import tech.pegasys.teku.api.schema.ProposerSlashing;
-import tech.pegasys.teku.beaconrestapi.handlers.AbstractHandler;
-import tech.pegasys.teku.beaconrestapi.schema.BadRequest;
-import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.beaconrestapi.MigratingEndpointAdapter;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.EndpointMetadata;
+import tech.pegasys.teku.infrastructure.restapi.endpoints.RestApiRequest;
+import tech.pegasys.teku.spec.datastructures.operations.ProposerSlashing;
 import tech.pegasys.teku.statetransition.validation.InternalValidationResult;
 import tech.pegasys.teku.statetransition.validation.ValidationResultCode;
 
-public class PostProposerSlashing extends AbstractHandler {
+public class PostProposerSlashing extends MigratingEndpointAdapter {
   public static final String ROUTE = "/eth/v1/beacon/pool/proposer_slashings";
   private final NodeDataProvider nodeDataProvider;
 
-  public PostProposerSlashing(final DataProvider dataProvider, final JsonProvider jsonProvider) {
-    this(dataProvider.getNodeDataProvider(), jsonProvider);
+  public PostProposerSlashing(final DataProvider dataProvider) {
+    this(dataProvider.getNodeDataProvider());
   }
 
-  public PostProposerSlashing(final NodeDataProvider provider, final JsonProvider jsonProvider) {
-    super(jsonProvider);
+  public PostProposerSlashing(final NodeDataProvider provider) {
+    super(
+        EndpointMetadata.post(ROUTE)
+            .operationId("postProposerSlashing")
+            .summary("Submit proposer slashing object")
+            .description(
+                "Submits proposer slashing object to node's pool and if passes validation node MUST broadcast it to network.")
```

Comment:
not new, but we should really say `and, if it passes validation, the node MUST...`